src/main/java/ca/trade/approach1/model/trade/AbstractTrade.java

```
@@ -0,0 +1,16 @@
+package ca.trade.approach1.model.trade;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+@ToString
+public abstract class AbstractTrade {
+    @NonNull
+    @Getter
+    private final Integer price;
```

если поле nonNull, то лучше int

---

Такая замена может привести к потенциальной проблеме. Когда экземпляр этого класса создан, а цена не задана, тогда она explicit будет выставлена в 0. Лучше fail fast, чем такое потенциально опасное поведение

---

можно проверять в конструкторе, какое значение приходит. И как можно создать екземляр, а цену не поставить, если поле final ?

---

проверил, да, так не выйдет. Заменил на int без NonNull